### PR TITLE
[RFC] Use module pattern with vim.lua and helper.lua

### PR DESCRIPTION
**Pull request by h-michael**
ref: #10724, #11276, https://github.com/neovim/neovim/pull/11276#issuecomment-545254058

### CODE REVIEW

File: test/functional/helpers.lua [hunk 1]
```diff
@@ -23,28 +23,28 @@ local sleep = global_helpers.sleep
...
 
 local start_dir = lfs.currentdir()
-module.nvim_prog = (
+helper.nvim_prog = (
   os.getenv('NVIM_PRG')
   or global_helpers.test_build_dir .. '/bin/nvim'
 )
 -- Default settings for the test session.
-module.nvim_set = (
+helper.nvim_set = (
```
Comment (justinmk) [inline test/functional/helpers.lua]: Let's not rename the `module.` cases until we think of a better name than "helper".

File: src/nvim/lua/vim.lua [hunk 2]
```diff
@@ -259,7 +260,6 @@ local function __index(t, key)
   end
 end
 
-
 -- vim.fn.{func}(...)
 local function fn_index(t, key)
```
Comment (justinmk) [inline src/nvim/lua/vim.lua]: this was renamed to `_fn_index` so there's a merge conflict.

File: test/functional/helpers.lua (hunk 1 above)
Comment (h-michael) [reply test/functional/helpers.lua]: Do you want to leave `test/helper.lua` unchanged this time as well?

Comment (blueyed) [reply test/functional/helpers.lua]: IIRC I've used `global_helpers` for `tests/helper.lua` initially (when adding the pattern) - not a much better name, but the idea was the it helps with generated tags.
So here `functional_helpers` could be used.
But I agree that "helper" is very generic.

**Comment by norcalli** [comment]
I accidentally did exactly this for vim.lua in #11442 , so we can close this.

**Pull request closed without merging**